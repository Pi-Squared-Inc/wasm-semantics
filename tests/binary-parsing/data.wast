(module $mymodule
  (memory 17)
  (data "data-passive")
  (data (offset (i32.const 1048576)) "data-active")
  (data (memory 0) (offset (i32.const 1048576)) "data-active-idx")
)
