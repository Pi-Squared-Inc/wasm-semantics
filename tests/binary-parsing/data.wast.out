<generatedTop>
  <k>
    #module (... types: .EmptyStmts , funcs: .EmptyStmts , tables: .EmptyStmts , mems: #memory (... limits: #limitsMin ( 17 ) , metadata:  )  .EmptyStmts , globals: .EmptyStmts , elem: .EmptyStmts , data: #data (... index: 0 , offset: .EmptyStmts , data: b"data-passive" )  #data (... index: 0 , offset: i32 . const 1048576  .EmptyStmts , data: b"data-active" )  #data (... index: 0 , offset: i32 . const 1048576  .EmptyStmts , data: b"data-active-idx" )  .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
